File: src/main/java/com/innercircle/project_one/survey/api/service/SurveyCreateService.java
package com.innercircle.project_one.survey.api.service;

import com.innercircle.project_one.survey.api.dto.SurveyDTO;
import com.innercircle.project_one.survey.api.repository.*;
import com.innercircle.project_one.survey.common.ApiResponse;
import com.innercircle.project_one.survey.common.SuccessResponse;
import com.innercircle.project_one.survey.domain.*;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@AllArgsConstructor
public class SurveyCreateService {

    private final SurveyRepository surveyRepository;
    private final SurveyVersionRepository surveyVersionRepository;
    private final SurveyService surveyService;


    @Transactional
    public ApiResponse saveSurvey(SurveyDTO surveyDTO) {
        Survey survey = createSurvey(surveyDTO);
        SurveyVersion newSurveyVersion = createSurveyVersion(survey);
        surveyService.saveSurveyObjects(survey, surveyDTO, newSurveyVersion);
        return new SuccessResponse<>("설문조사 폼이 저장되었습니다.");
    }


    private Survey createSurvey(SurveyDTO surveyDTO) {
        return surveyRepository.save(new Survey(surveyDTO.title(), surveyDTO.description()));
    }

    private SurveyVersion createSurveyVersion(Survey survey) {
        return surveyVersionRepository.save(new SurveyVersion(1L, survey));
    }


}

File: src/main/kotlin/org/inner/circle/survey/application/service/SurveyCreateService.kt
package org.inner.circle.survey.application.service

import io.github.oshai.kotlinlogging.KotlinLogging
import org.inner.circle.survey.domain.storage.SurveyStorage
import org.inner.circle.survey.presentation.dto.SurveyCreateRequest.CreateSurvey
import org.springframework.stereotype.Service

@Service
class SurveyCreateService(
    private val surveyStorage: SurveyStorage,
) {
    private val log = KotlinLogging.logger {}

    fun saveSurvey(createSurvey: CreateSurvey) {
        log.info { "Creating survey with title: ${createSurvey.title}" }
        surveyStorage.save(createSurvey)
    }
}
